PokeDemoTests/PokemonListPresenterTests.swift

```diff
@@ -0,0 +1,142 @@
+        
+        // When
+        presenter = PokemonListPresenter(service: mockService)
+        presenter.viewDidLoad()
+        
+        // Then
+        XCTAssertEqual(mockService.sendRequestWithJSONIsCalledIndex, 1)
+        XCTAssertEqual(mockService.sendRequestWithJSONEndPoint, "https://pokeapi.co/api/v2/pokemon")
+    }
+    
+    func testGetPokemonListReturnsSuccessFromAPI(){
+        // Given
+        let mockService = MockService()
+        let mockDelegate = MockPresenterDelegate()
+        presenter = PokemonListPresenter(service: mockService)
+        presenter.setViewDelegate(delegate: mockDelegate)
+        presenter.viewDidLoad()
+        
+        // When service sends successful response
+        let stubPokemons = [Pokemon.init(name: "test", url: "Test URL")]
+//        mockService.onCompletion?(stubPokemons, nil)
+        
+        // Then
+        XCTAssertEqual(mockDelegate.pokemons, stubPokemons)
+        // Check the mockDelegate showPokemonList function is called with right Pokemon array
+    }
+    
+    func testGetPokemonListReturnsErrorFromAPI(){
+        // Given
+        let mockService = MockService()
+        let mockDelegate = MockPresenterDelegate()
+        presenter = PokemonListPresenter(service: mockService)
+        presenter.setViewDelegate(delegate: mockDelegate)
+        presenter.viewDidLoad()
+        
+        // When service sends successful response
+        mockService.onCompletion?(nil, TestError())
+        
+        // Then
+        XCTAssertEqual(mockDelegate.showAlertCalled, true)
+        XCTAssertEqual(mockDelegate.showAlertTitle, Constants.error)
+        XCTAssertEqual(mockDelegate.showAlertMessage, "The operation couldn’t be completed. (PokeDemoTests.TestError error 1.)")
+        // Check the mockDelegate showPokemonList function is called with right Pokemon array
+    }
+    
+    func testGetPokemonListReturnsNoInternetFailureFromAPI(){
+        // Given
+        let mockService = MockService()
+        let mockDelegate = MockPresenterDelegate()
+        presenter = PokemonListPresenter(service: mockService)
+        presenter.setViewDelegate(delegate: mockDelegate)
+        presenter.viewDidLoad()
+        
+        // When service sends successful response
+        mockService.onCompletion?(nil, TestError())
+        
+        // Then
+        XCTAssertEqual(mockDelegate.showAlertCalled, true)
+        XCTAssertEqual(mockDelegate.showAlertTitle, Constants.error)
+        XCTAssertEqual(mockDelegate.showAlertMessage, Internet.shared.noInternet)
+    }
+    
+    func testGetPokemonListReturnsInvalidDataFromAPI(){}
+    
+    func testGetPokemonListReturnsGeneralErrorFromAPI(){
+        // Given
+        let mockService = MockService()
+        
+        // When
+        presenter = PokemonListPresenter(service: mockService)
+        presenter.viewDidLoad()
+        
+        // Then
+        XCTAssertEqual(mockService.sendRequestWithJSONIsCalledIndex, 1)
+    }
```

Comment:
Remove this test

Reply:
Covered in a7a16501021c48b8a1bbbf9c8a519a0146eeeb8c